src/Timers.cpp

```diff
@@ -1179,82 +1151,82 @@ bool Timers::TimerUpdatesRegular()
-          m_timers[i].strPlot = newtimers[j].strPlot;
-          m_timers[i].iChannelId = newtimers[j].iChannelId;
-          m_timers[i].strChannelName = newtimers[j].strChannelName;
-          m_timers[i].startTime = newtimers[j].startTime;
-          m_timers[i].endTime = newtimers[j].endTime;
-          m_timers[i].iWeekdays = newtimers[j].iWeekdays;
-          m_timers[i].iEpgID = newtimers[j].iEpgID;
-          m_timers[i].tags = newtimers[j].tags;
+          newTimer.iUpdateState = VU_UPDATE_STATE_UPDATED;
+          existingTimer.iUpdateState = VU_UPDATE_STATE_UPDATED;
+          existingTimer.strTitle = newTimer.strTitle;
+          existingTimer.strPlot = newTimer.strPlot;
+          existingTimer.iChannelId = newTimer.iChannelId;
+          existingTimer.strChannelName = newTimer.strChannelName;
+          existingTimer.startTime = newTimer.startTime;
+          existingTimer.endTime = newTimer.endTime;
+          existingTimer.iWeekdays = newTimer.iWeekdays;
+          existingTimer.iEpgID = newTimer.iEpgID;
+          existingTimer.tags = newTimer.tags;
 
           iUpdated++;
         }
       }
     }
   }
 
-  unsigned int iRemoved = 0;
+  //Remove any timers that are no longer valid
+  unsigned int iRemoved = m_timers.size();
 
-  for (unsigned int i=0; i<m_timers.size(); i++)
-  {
-    if (m_timers.at(i).iUpdateState == VU_UPDATE_STATE_NONE)
-    {
-        XBMC->Log(LOG_INFO, "%s Removed timer: '%s', ClientIndex: '%d'", __FUNCTION__, m_timers.at(i).strTitle.c_str(), m_timers.at(i).iClientIndex);
-        m_timers.erase(m_timers.begin()+i);
-        i=0;
-        iRemoved++;
-    }
-  }
+  m_timers.erase(
+    std::remove_if(m_timers.begin(), m_timers.end(), 
+      [](const Timer& timer) { return timer.iUpdateState == VU_UPDATE_STATE_NONE; }), 
+    m_timers.end());
+
+  iRemoved -= m_timers.size();
+
+  //Add any new autotimers
   unsigned int iNew=0;
 
-  for (unsigned int i=0; i<newtimers.size();i++)
+  for (auto& newTimer : newtimers)
   { 
-    if(newtimers.at(i).iUpdateState == VU_UPDATE_STATE_NEW)
+    if(newTimer.iUpdateState == VU_UPDATE_STATE_NEW)
     {  
-      Timer &timer = newtimers.at(i);
-      timer.iClientIndex = m_iClientIndexCounter;
-      timer.strChannelName = vuData.GetChannels().at(timer.iChannelId-1).strChannelName;
-      XBMC->Log(LOG_INFO, "%s New timer: '%s', ClientIndex: '%d'", __FUNCTION__, timer.strTitle.c_str(), m_iClientIndexCounter);
-      m_timers.push_back(timer);
+      newTimer.iClientIndex = m_iClientIndexCounter;
+      newTimer.strChannelName = vuData.GetChannels().at(newTimer.iChannelId-1).strChannelName;
+      XBMC->Log(LOG_INFO, "%s New timer: '%s', ClientIndex: '%d'", __FUNCTION__, newTimer.strTitle.c_str(), m_iClientIndexCounter);
+      m_timers.emplace_back(newTimer);
       m_iClientIndexCounter++;
       iNew++;  
     } 
   }  
 
-  for (unsigned int i=0; i<m_timers.size();i++)
+  //Link any Readonly timers to Repeating Timers
+  for (auto& repeatingTimer : m_timers)
```

Comment:
this imo could be `const auto&`

Reply:
It can be once the argument to isChildOfParent is const which it can be.